### ISSUE TITLE
Erroneous underscores in templates with templates

### ISSUE THREAD

**Issue by astrelsky** [NationalSecurityAgency/ghidra#4081]
**Describe the bug**
Analyzing a program with debug info (both dwarf and pdb) that contains a symbol with a template containing a template will apply a symbol which does not match the symbol in the debug info.

**To Reproduce**
1. Compile the code under *attachments* with optimizations disabled to prevent inlining.
2. Go to the main function and see `Holder<Holder<int>_>::Holder` instead of `Holder<Holder<int>>::Holder`

**Expected behavior**
There should obviously be no underscored where there were none to begin with.

**Attachments**
```c++
template <typename T>
class Holder {
    T t;
public:
    Holder() : t(){}
    ~Holder() = default;
    T get() const { return t; }
};

int main() {
    Holder<Holder<int>> var;
    return var.get().get();
}
```

**Environment (please complete the following information):**
 - OS: Microsoft Windows [Version 10.0.22000.556]
 - Java Version: 11
 - Ghidra Version: 10.1.2 and 10.2_DEV 21aa55cf2dd173c668ca042cf8d352fb8462394c

**Additional context**
Sometimes a demangler will insert a space for readability. Removing it instead of replacing it with an underscore would be correct.

**Comment by astrelsky** [comment]
@ryanmkurtz this also occurs with DWARF import and PDB loading.

**Comment by ghizard** [comment]
This is a known issue.  The issue is not really either Demangler or PDB/DWARF but that Ghidra does not allow spaces in certain things such as symbols and namespaces.  There is no time line for when a solution for this is expected.

**Comment by astrelsky** [comment]
> This is a known issue. The issue is not really either Demangler or PDB/DWARF but that Ghidra does not allow spaces in certain things such as symbols and namespaces. There is no time line for when a solution for this is expected.

Erroneous whitespace characters where there were none should be replaced with nothing and not an underscore. I believe it's just due to misuse of `SymbolUtilities.replaceInvalidChars` in certain situations and not on the grand scale of things.

**Comment by ghizard** [comment]
Yes.  The "known issue" is "an error," and that is the subject code that replaces the characters, but supposedly to ensure the underlying problems do not get triggered.

I make the exact same arguments to others on the team; I feel that someone or people on the team who did not know languages well made that decision, and that is exactly what I was repeating to someone on the team yesterday.

It is related to your #1175.

I'd like to fix it or have others fix it, but there are some underlying issues that are being worked on that have bearing on this issue.

**Comment by astrelsky** [comment]
> Yes. The "known issue" is "an error," and that is the subject code that replaces the characters, but supposedly to ensure the underlying problems do not get triggered.
> 
> I make the exact same arguments to others on the team; I feel that someone or people on the team who did not know languages well made that decision, and that is exactly what I was repeating to someone on the team yesterday.
> 
> It is related to your #1175.
> 
> I'd like to fix it or have others fix it, but there are some underlying issues that are being worked on that have bearing on this issue.

That's going pretty far back now. I mistook your initial response as a *won't fix* and am willing to put up a fight about it. It may seem like a stupid and minor problem but when having to manually rename 100s of labels so that the correct `this` type gets used, as well as so it is correct, gets painfully tedious.
